### ISSUE TITLE
[Fwdproxy] Apply GitHub action after each merge to main

### ISSUE THREAD

**Issue by npeterson54** [department-of-veterans-affairs/va.gov-team#57551]
## Description
Apply terraform changes to live environments (dev, staging, sandbox, utility, production), while ephemeral environments are left to manually trigger the action to receive their updates.

- [ ] Depends on #76237 

## Resources 
- Some examples might be (using a base action with several "trigger actions"):
  - https://github.com/department-of-veterans-affairs/devops/blob/master/.github/workflows/packer-build-platform-base-weekly.yml
  - https://github.com/department-of-veterans-affairs/devops/blob/master/.github/workflows/packer-build-platform-base.yml

## Acceptance Criteria
- [ ] Upon merge to `main` terraform changes are planned and applied to appropriate live environments automatically
- [ ] A manual triggering of the action will update a specified ephemeral environment
- [ ] Create/update Documentation

## Refinement Guidance - Check the following before working on this issue: 
- [x] _Team label assigned ("platform-tech-team-2")_
- [x] _Epic assigned (if needed)_ 
- [x] _Estimated (points assigned)_
- [x] _Sprint assigned (once planned)_
- [x] _Team member(s) assigned_

**Comment by npeterson54** [comment]
Was not able to complete in sprint, moving into next sprint.  This will be our sprint goal

**Comment by npeterson54** [comment]
Rolling into next sprint due to some PTO

**Comment by EWashb** [comment]
@npeterson54 I see this one has been labeled a sprint goal for a few sprints. Is there a blocker you're facing on this that I can help resolve? Also, is it expected to be complete in the current sprint?

**Comment by npeterson54** [comment]
No, no blockers, just some PTO with the person that was working on it that we didn't expect.  It should be done this sprint @EWashb

**Comment by npeterson54** [comment]
Did not quite complete, moving into next sprint

**Comment by ph-One** [comment]
This work needs some R&D. It should be removed from the current and future sprint until said R&D/Spike has been done. It's not entirely clear how this would work, considering we're doing blue/green and there will be bouts of logic and naming standards to check on.

**Comment by ph-One** [comment]
The necessary Actions were recently created [this sprint] to support this work

**Comment by EWashb** [comment]
@npeterson54 and @ph-One thanks for the update. Sounds like you already created the other tickets that you need to complete first? Nate, can you add those tickets as blocking this ticket just so we know how that is related?

**Comment by ph-One** [comment]
Checking on the tickets @EWashb . I don't recall if separate tickets were created, or if the work was done thinking it'd be under this ticket -- they should be separate tickets. If the tickets are not present, I'll create them and link them here

**Comment by ph-One** [comment]
https://github.com/department-of-veterans-affairs/va.gov-team/issues/61129

**Comment by EWashb** [comment]
Thanks for clarifying @ph-One

**Comment by ph-One** [comment]
Moving to backlog since this needs fleshed out a bit more. There are behaviors for releases that are not defined well enough, such as making sure that lower tiers are deployed and confirmed before higher tiers, and enforcing this through Actions.

**Comment by npeterson54** [comment]
Adding the needs-refinement label back on

**Comment by ph-One** [comment]
@barbarello @stoiven Based on what I see in the forward proxy repo, this work has been completed. Please close this ticket if it is no longer needed.


- https://github.com/department-of-veterans-affairs/vsp-platform-fwdproxy/actions/workflows/deploy-daily.yaml
- https://github.com/department-of-veterans-affairs/vsp-platform-fwdproxy/actions/workflows/deploy-continuous.yaml
- https://github.com/department-of-veterans-affairs/vsp-platform-fwdproxy/actions/workflows/deploy-continuous-tf.yaml

**Comment by ph-One** [comment]
This work has been completed. See above comment for references to the implemented GitHub Actions that close this work item out.
